test/core/test_lwt_sequence.ml

```diff
@@ -190,4 +230,144 @@ let suite = suite "lwt_sequence" [
     | None -> Lwt.return_false
     | Some v -> Lwt.return (5 = v)
   end;
+
+  test "iter_l Empty" begin fun () ->
+    test_iter Lwt_sequence.iter_l empty_array (Lwt_sequence.create ())
+  end;
+
+  test "iter_l" begin fun () ->
+    test_iter Lwt_sequence.iter_l l_filled_array (filled_sequence ())
+  end;
+
+  test "iter_r Empty" begin fun () ->
+    test_iter Lwt_sequence.iter_r empty_array (Lwt_sequence.create ())
+  end;
+
+  test "iter_r" begin fun () ->
+    test_iter Lwt_sequence.iter_r r_filled_array (filled_sequence ())
+  end;
+
+  test "iter_node_l Empty" begin fun () ->
+    test_iter_node Lwt_sequence.iter_node_l empty_array (Lwt_sequence.create ())
+  end;
+
+  test "iter_node_l" begin fun () ->
+    test_iter_node Lwt_sequence.iter_node_l l_filled_array (filled_sequence ())
+  end;
+
+  test "iter_node_r Empty" begin fun () ->
+    test_iter_node Lwt_sequence.iter_node_r empty_array (Lwt_sequence.create ())
+  end;
+
+  test "iter_node_r" begin fun () ->
+    test_iter_node Lwt_sequence.iter_node_r r_filled_array (filled_sequence ())
+  end;
+
+  test "iter_node_l with removal" begin fun () ->
+    test_iter_rem Lwt_sequence.iter_node_l l_filled_array (filled_sequence ())
+  end;
+
+  test "iter_node_r with removal" begin fun () ->
+    test_iter_rem Lwt_sequence.iter_node_r r_filled_array (filled_sequence ())
+  end;
+
+  test "fold_l" begin fun () ->
+    let sum = Lwt_sequence.fold_l (fun v e -> v + e) (filled_sequence ()) 0 in
+    Lwt.return (sum = 0)
```

Comment:
Totally a nit, but I would change `+` to `*` (easiest) or change the sequence elements. Starting with zero and checking for zero could fail to catch a mistake where we accidentally forget to call `f` at all, and just propagate the accumulator during `fold`. Not that anyone is actually going to edit `Lwt_sequence` and do that, so feel free not to fix it :)